lib/views/add_shoe/add_shoe_view_model.dart

```diff
@@ -1,24 +1,85 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_shoe_app/app/app.locator.dart';
+import 'package:flutter_shoe_app/views/home/shoe_object.dart';
+import 'package:get/get.dart';
 import 'package:image_picker/image_picker.dart';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+import '../../core/services/api/api_service.dart';
 
 class AddShoeViewModel extends ChangeNotifier {
+  TextEditingController nameController = TextEditingController();
+  TextEditingController descriptionController = TextEditingController();
+  TextEditingController priceController = TextEditingController();
+  TextEditingController categoryController = TextEditingController();
+
   final imagePicker = ImagePicker();
+  String imagePath = "";
   List<XFile>? images = [];
   List<String> selectedSizes = [];
+  List<String> base64String = [];
 
   Future getImages() async {
     List<XFile>? selectedImages = await imagePicker.pickMultiImage();
-    if (selectedImages != null && selectedImages.isNotEmpty == true) {
+    if (selectedImages != null && selectedImages.isNotEmpty) {
+      for (int i = 0; i < selectedImages.length; i++) {
+        File imageFile = File(selectedImages[i].path);
+        imagePath = imageFile.path;
+        //final String jsonImage = jsonEncode(imagePath);
+        //print(jsonImage);
+
+        File imagefile = File(imagePath); //convert Path to File
+        Uint8List imagebytes = await imagefile.readAsBytes(); //convert to bytes
+
+        String base64img =
+            'data:image/png;base64,${base64.encode(imagebytes)}'; //convert bytes to base64 string
+        //print(base64img);
+
+        base64String.add(base64img);
+
+        // for (String s in base64String) {
+        //   print(s);
+        // }
+      }
       images!.addAll(selectedImages);
+    } else {
+      print('No images Selected');
     }
-    // else {
-    //   print('No images Selected');
-    // }
     notifyListeners();
   }
 
   void onSizeSelectedCallback(List<String> values) {
     selectedSizes = values;
+    //print(selectedSizes);
     notifyListeners();
   }
+
+  final ApiService apiService = locator<ApiService>();
+
+  //adding new shoe item
+  Future<void> addShoes(
+    BuildContext context,
```

Comment:
Rename method name from addShoes to addShoe because the method will only add 1 shoe at a time.